cpp/dolfin/mesh/cell_types.cpp

```diff
@@ -558,10 +587,13 @@ int mesh::cell_degree(mesh::CellType type, int num_nodes)
     else
       throw std::runtime_error("Higher order tetrahedron not supported");
   case mesh::CellType::quadrilateral:
-    if (num_nodes == 4)
-      return 1;
-    else
-      throw std::runtime_error("Higher order quadrilateral not supported");
+  {
+    int i = sqrt(num_nodes);
+    if (num_nodes == i * i)
+      return i - 1;
+    throw std::runtime_error("Quadrilateral of order "
+                             + std::to_string(num_nodes) + " not supported");
```

Comment:
Message could be better. If `num_nodes != i*i`, then it is not just the wrong order, it is a different element type, probably.